### PR TITLE
Font Library: Refactor stylesheet using CSS variables

### PR DESCRIPTION
**Pull request by t-hamano**
Mentioned in this comment: https://github.com/WordPress/gutenberg/pull/58117#issuecomment-1907976999

## What?

This PR refactors hard-coded color, padding, margin, etc. values into CSS variables in the stylesheet that defines the styles of the font library.

> [!NOTE]
> This PR might have a selector conflict with #58221, which removes header/main/footer elements. If #58221 was merged first, we will need to rebase on the latest trunk before merging this PR.

| Before | After |
|--------|--------|
| image| image|
|  image| image |
|  image| image| 

## Why?

My understanding is that colors and spaces need to follow the rules defined as variables as much as possible.

## How?

Regarding the space, I completely kept the current value and replaced it with a CSS variable (`$grid-unit-{XX}`) .

Regarding the color, I replaced it with a CSS variable that is as close as possible to the color and matches the context, as shown below.

| Color (trunk) | Color (this PR)                                 |
| ------------- | ----------------------------------------------- |
| `#6e6e6e`     | `$gray-700` (`#757575`)                         |
| `#e5e5e5`     | `$gray-300` (`#dddddd`) or `$gray-200` (`#e0e0e0`) |
| `#e6e6e6`     | `$gray-300` (`#dddddd`)                            |
| `#f0f0f0`     | `$gray-100` (`#f0f0f0`)                         |
| `#ffffff`        | `$white` (`#ffffff`)                               |

## Testing Instructions

Make sure there are no unnatural spaces or colors in the Font Library.

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss [hunk 1]
```diff
@@ -50,41 +50,40 @@
...
-	padding-bottom: 1rem;
+	border-bottom: 1px solid $gray-200;
+	padding-bottom: $grid-unit-20;
 }
 
 .font-library-modal__tabs {
 	[role="tablist"] {
 		position: sticky;
 		top: 0;
-		width: calc(100% + 64px);
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/style.scss:83]: There is a negative margin on the left and right, so there is no need to calculate the width.

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss [hunk 2]
```diff
@@ -93,35 +92,27 @@
...
 	}
 }
 
-.font-library-modal__font-name {
-	font-weight: bold;
-}
-
-.font-library-modal__font-filename {
-	color: #6e6e6e;
-}
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/style.scss:123]: It seems that there are no elements with these selectors.

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss [hunk 3]
```diff
@@ -154,7 +145,3 @@ button.font-library-modal__upload-area {
 		max-width: 350px;
 	}
 }
-
-.font-library-modal__font-collection__notice {
-	margin: 0;
-}
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/style.scss:160]: The Notice component's margins have now been removed, so we no longer need to reset them.

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss [hunk 4]
```diff
@@ -93,35 +92,27 @@
 	align-items: center;
 	display: flex;
 	justify-content: center;
-	height: 250px;
+	height: $grid-unit-80 * 4; // 256px
```
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/style.scss:95]: This is the height of the area where we will upload fonts. This is the only exception, and I adjusted it to a multiple of 8.

upload-area

**Review by jasmussen — APPROVED** [review]
Thank you!

**Comment by cbravobernal** [comment]
I just cherry-picked this PR to the release/17.6 branch to get it included in the next release: https://github.com/WordPress/gutenberg/commit/3b8f1c1451098e541d272090bcb6324edf92080e

**Pull request merged**